File: src/storage/Node.cpp
#include <QDebug>
#include <iostream>
#include "Node.hpp"

Node::Node()
{

}

void Node::from_json(const QJsonObject &data, int re)
{
	qDebug() << "node from_json, re: " << re;

	if(data.contains("name")){
		this->name = data.value("name").toString();
	}

	if(data.contains("childrens") && data["childrens"].isArray()){

		QJsonArray chnodes = data["childrens"].toArray();

		for(auto chnode : chnodes){
			auto chq = chnode.toObject();

			Node n = Node();
			n.from_json(chq, re);
			this->childrens.append(n);
		}
	}
}

void Node::reprint(const uint level)
{

	for(int i=0; i<level; i++){
		std::cout << "    ";
	}

	std::cout << this->name.toStdString() << std::endl;

	for(auto ch: this->childrens){
		ch.reprint(level + 1);
	}
}

File: py/storage/Node.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

class Node(object):
	def __init__(self):
		self.name = ""
		self.childrens = []
		
		
	def parse(self, data: dict, re: bool = False):
		self.name = data.get("name", "")
		
		
		
		childrens = data.get("childrens", [])
		for ch in childrens:
			n = Node()
			n.parse(ch, re)
			self.childrens.append(n)
			
			
	def reprint(self, level = 0):
		
		print("--"*level + self.name)
		for ch in self.childrens:
			ch.reprint(level + 1)
			